src/litdata/utilities/subsample.py

```diff
@@ -0,0 +1,172 @@
+from typing import Any, Dict, List, Optional, Tuple
+
+import numpy as np
+
+
+def shuffle_lists_together(
+    list1: List[Any], list2: List[Any], random_seed_sampler: Optional[np.random.RandomState] = None
+) -> Tuple[List[Any], List[Any]]:
+    """Shuffles list1 and applies the same shuffle order to list2.
+
+    Args:
+        list1: The first list to shuffle.
+        list2: The second list to shuffle in correspondence with list1.
+        random_seed_sampler: Random seed sampler to be used for shuffling
+
+    Returns:
+        A tuple containing the shuffled versions of list1 and list2.
+
+    """
+    # Sanity check
+    if len(list1) != len(list2):
+        raise ValueError("Lists must be of the same size.")
+
+    if random_seed_sampler is None:
+        random_seed_sampler = np.random.RandomState([42])
```

Comment:
Let's pass down the seed as an argument.

Reply:
Is this what you mean:
```python
def shuffle_lists_together(
    list1: List[Any], list2: List[Any], random_seed_sampler: Optional[np.random.RandomState] = None, seed: Optional[int] = None
) -> Tuple[List[Any], List[Any]]:
 
    if random_seed_sampler is None and seed is None:
        raise ValueError("Both random_seed_sampler and seed are none.")

    if random_seed_sampler is None and seed is None:
        raise ValueError("Both random_seed_sampler and seed have been provided. Provide either one of them only.")

    # Sanity check
    if len(list1) != len(list2):
        raise ValueError("Lists must be of the same size.")

    if random_seed_sampler is None:
        random_seed_sampler = np.random.RandomState([seed])
    # ........ rest all fine
```

Reply:
I think seed: Optional[int] = None is enough